services/gateway/gateway-service.mjs

```diff
@@ -272,26 +277,40 @@ class GatewayService {
     return true;
   }
 
-  async broadcastResetUser(socket , uid, x, y, facing){
-    socket.broadcast.emit("location",{uid:uid,x:x,y:y,facing:facing});
-  }
+  // async broadcastResetUser(socket , uid, x, y, facing){
+  //   socket.broadcast.emit("location",{uid:uid,x:x,y:y,facing:facing});
+  // }
 
-  getLastPosition(uid){
-    assert.fail('Not implemented');
-    return {x:0,y:0,facing:'up'}
+  _checkPositionEmpty(x,y){
+    return gameMap.getCell(x,y);
   }
 
-  updatePosition(uid,x,y,facing){
-    assert.fail('Not implemented');
+  _getDistanceSquare(a,b){
+    return Math.pow(Math.abs(a.x - b.x),2) + Math.pow(Math.abs(a.y - b.y),2);
   }
 
-  getDistanceSquare(a,b){
-    return Math.abs(a.x - b.x)^2 + Math.abs(a.y - b.y)^2
+  _borderCheck(msg,mapSize,mapShift){
+    if(msg.x > mapSize.width + mapShift.x || msg.x < mapShift.x){
+      return false;
+    }
+    if(msg.y > mapSize.height + mapShift.y || msg.y < mapShift.y){
+      return false;
+    }
+    return true;
   }
 
-  checkPositionEmpty(x,y){
-    return gameMap.getCell(x,y);
+  _nearByGridCheck(msg,lastRecord){
+    let lastPosition = {x:lastRecord.x , y:lastRecord.y};
+    let targetPosition = {x:msg.x , y:msg.y};
+    let distanceSquire = this._getDistanceSquare(lastPosition,targetPosition);
+    let speed = 1 ;
```

Comment:
If we are dealing with speed, we need to take the time (duration between 2 moves) into consideration as well.